File: app/models/preferences.py
"""
用户偏好设置模型

单用户本地部署模式下，偏好设置存储在 system_configs 集合中，
不再绑定用户账户。
"""

from typing import List
from pydantic import BaseModel, Field


class UserPreferences(BaseModel):
    """用户偏好设置"""

    # 分析偏好
    default_market: str = "A股"
    default_depth: str = "3"  # 1-5级，3级为标准分析（推荐）
    default_analysts: List[str] = Field(default_factory=lambda: ["市场分析师", "基本面分析师"])
    auto_refresh: bool = True
    refresh_interval: int = 30  # 秒

    # 外观设置
    ui_theme: str = "light"
    sidebar_width: int = 240

    # 语言和地区
    language: str = "zh-CN"

    # 通知设置
    notifications_enabled: bool = True
    email_notifications: bool = False
    desktop_notifications: bool = True
    analysis_complete_notification: bool = True
    system_maintenance_notification: bool = True

File: frontend/src/types/preferences.ts
// 偏好设置（单用户本地部署模式，无账号概念）
export interface UserPreferences {
  // 分析偏好
  default_market: 'A股' | '美股' | '港股'
  default_depth: '1' | '2' | '3' | '4' | '5'  // 1-5级分析深度
  default_analysts?: string[]  // 默认分析师列表：市场分析师、基本面分析师、新闻分析师
  auto_refresh?: boolean
  refresh_interval?: number

  // 外观设置
  ui_theme: 'light' | 'dark' | 'auto'
  sidebar_width?: number

  // 语言和地区
  language: 'zh-CN' | 'en-US'

  // 通知设置
  notifications_enabled: boolean
  email_notifications: boolean
  desktop_notifications?: boolean
  analysis_complete_notification?: boolean
  system_maintenance_notification?: boolean
}
